### PR TITLE
Add `user` flag to nbextension

### PR DESCRIPTION
**Pull request by francisco-dlp**
Add --user flag to nbextension as somehow suggested by @hainm in #170.

### CODE REVIEW

File: ipyparallel/apps/ipclusterapp.py [hunk 1]
```diff
@@ -572,35 +572,42 @@ def start():
...
     """
-    
+
     examples = """
     ipcluster nbextension enable
     ipcluster nbextension disable
     """
     version = __version__
-    
+    user = Bool(False, "Apply the operation only for the given user").tag(
```
Comment (Carreau) [inline ipyparallel/apps/ipclusterapp.py]: You have to specify `help=' explicitly I think, the second positional argument is `allow_none` for `Bool`.

File: ipyparallel/apps/ipclusterapp.py [hunk 2]
```diff
@@ -572,35 +572,42 @@ def start():
...
+        config=True)
+    # flags = {}
```
Comment (Carreau) [inline ipyparallel/apps/ipclusterapp.py]: do you want to remove this commented line ?

**Comment by Carreau** [comment]
Hi @francisco-dlp thanks a lot !

You have your "autostrip trailing whitespace" of your editor likely turned on. If you can configure it on a per-project basis to not do that we would appreciate; otherwise it adds a lot of churn to the codebase, and often make git-blame more complicated to use. 

It's ok if you've already committed it though !

File: ipyparallel/apps/ipclusterapp.py [hunk 3]
```diff
@@ -572,35 +572,42 @@ def start():
...
+    flags = Dict(dict(user= (
+        {'IPClusterNBExtension': {'user': True}},
+        "Apply the operation only for the given user",
+    )))
```
Comment (Carreau) [inline ipyparallel/apps/ipclusterapp.py]: I'm confused as of why you mix dict literals with calling `dict`, is there a reason ?

File: ipyparallel/nbextension/install.py [hunk 4]
```diff
@@ -8,23 +8,23 @@
...
     if V(notebook.__version__) < V('4.2'):
         return _install_extension_nb41(enable)
-    
+
     from notebook.nbextensions import install_nbextension_python, enable_nbextension, disable_nbextension
     from notebook.serverextensions import toggle_serverextension_python
-    toggle_serverextension_python('ipyparallel.nbextension')
-    install_nbextension_python('ipyparallel')
+    toggle_serverextension_python('ipyparallel.nbextension', user=user)
+    install_nbextension_python('ipyparallel', user=True)
```
Comment (minrk) [inline ipyparallel/nbextension/install.py]: user=user here?

**Comment by francisco-dlp** [comment]
Thanks @Carreau  and @minrk  for the review. All should be fixed now except for the removal of the trailing spaces. As only empty lines are affected, I bet that they're harmless, but, if you disagree I am happy to put back the spaces.

**Comment by minrk** [comment]
Thanks! Don't worry about the whitespace changes here, but it would be great if committing those changes could be avoided in the future.

**Pull request merged**